Lib/webbrowser.py

```diff
@@ -484,6 +485,12 @@ def register_X_browsers():
 
 # Prefer X browsers if present
 if os.environ.get("DISPLAY"):
+    try:
+        cmd = "xdg-settings get default-web-browser".split()
+        _preferred_browser = subprocess.check_output(cmd).decode().strip()
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        pass
```

Comment:
You could use shutil.which to determine if xdg-settings exists before trying to call it (LBYL).  But here ignoring the exception (EAFP) seems right to me.